build/Dockerfile

```diff
@@ -1,14 +1,17 @@
 # ---
 # Stage 1: Install certs, build binary, create default config file
 # ---
-FROM ghcr.io/project-zot/golang:1.20 AS builder
+FROM --platform=$BUILDPLATFORM ghcr.io/project-zot/golang:1.20 AS builder
+
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETPLATFORM
```

Comment:
I see TRAGETPLATFORM and a BUILDTLATFORM variables.
What is TRAGETPLATFORM used for, and how are we passing BUILDPLATFORM?
Didn't we need changes in the makefile?

Reply:
They are passed by Docker/Buildkit. Buildplatform is where the image is being built (otherwise the builder would try emulating the image for the target platform), Targetplatform is well....the target platform used for cross-compilation.

No need to pass these build args manually.

Reply:
I guess it can be removed, I don't see it used either.

Reply:
Looks ok, if it builds locally for your use case, I agree with merging this.